code-challenges.test.js

```diff
@@ -24,11 +24,40 @@ var secretCodeWord2 = "Gobbledygook"
 var secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("code", () => {
+    test("returns a coded message", () => {
+      expect(code(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(code(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(code(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
 
 
 // b) Create the function that makes the test pass.
 
-
+// create function called letterValue
+// parameter = string
+// create a variable that converts certain letters to numbers using colon to define a property
+//if statement (if the length of the string is 1, return the string with the new assignments from the number variable)
+// return the string with toLowerCase to eliminate any uppercase letters
+//split the string into individual strings to itterate through
+//map through each string
+//join the string back into one.
+
+const letterValue = (string) => {
+    var number={
+        a: 4, e: 3, i: 1, o: 0
+    }
+    if(string.length == 1) return number[string] || string;
+    return string.toLowerCase().split('').map(letterValue).join('');
```

Comment:
I'm gonna be honest, I'm at a loss for how this is working at all. letterValue inside of map should need more arguments to accomplish what it is accomplishing, or so I thought.

Reply:
Okay playing around with it something is weird about "o." Also very creative way to solve this problem; Though I would like some clarity on how it works.